### DISCUSSION TITLE
Examples for deleting, rescheduling and updating a scheduled job

### DISCUSSION THREAD

**Discussion by IgnisDa** [apalis-dev/apalis#142] [Q&A] [answered]
I searched all the examples in this repo but could not find any sample code to `kill`, `reschedule` and `update_by_id`. 

My use case is that I want to schedule a job and then edit the time it needs to run later (I think this should be `reschedule`?). I would also like to know how to change the `information` parameter sent to a scheduled job (I think it's the `update_by_id` method). Finally, another method to completely kill a scheduled job.

**Comment by IgnisDa** [comment]
~I figured out `.update_by_id`~.

For `.reschedule`, what does the `wait` argument actually do? And I have no idea what to put in `worker_id` argument for the `.kill` method.

**Comment by IgnisDa** [comment]
As an API consumer, I think it would be great if all these methods just take in `JobId` as input.

**Comment by geofmureithi** [comment]
The `reschedule` method  takes in the job and the amount of time to wait until its back into the queue.
The `WorkerId` is just the name of the worker eg `email-worker-1`
To kill a job is to tell a storage that the worker rejected a job. Currently you cant kill a job outside of its function signature because you have to get the `WorkerId` from context.

That mentioned, there is a reason apalis discourages using these apis directly. For once they  can create undefined behavior since they are storage specific rather than worker specific. ( Think worker continues to execute a killed job)

If you have logic you want eg to kill a job if it returns a specific error then you should use layers, because they are predictable,testable etc.

If you need you go lower than that I advice you get the connection pool and write some sql or redis commands.

I am curious though what you are unable to do currently.

**Comment by IgnisDa** [comment]
As I mentioned, this is what I want to do:

> My use case is that I want to schedule a job and then edit the time it needs to run later (I think this should be reschedule?). I would also like to know how to change the information parameter sent to a scheduled job (I think it's the update_by_id method). Finally, another method to completely kill a scheduled job.

**Comment by geofmureithi** [comment]
When you push/schedule a job, you get a JobId,
To edit the time it runs, just use update_by_id and update run at.
To kill a running job, you need to be inside that job function (ideally) eg by an early return.
To kill a scheduled job that is not yet running ie still in queue, just use update_by_id, and set status to killed.
You should be able to do these fairly straight forward.
I will encourage you look at the actual storage to understand what is happening in those calls.

**Comment by IgnisDa** [comment]
> To kill a scheduled job that is not yet running ie still in queue, just use update_by_id, and set status to killed.

Can you give an example, I was not able to figure this out.

**Comment by geofmureithi** [comment, marked as answer]
If you have the job
- Use set status: https://docs.rs/apalis/latest/apalis/prelude/struct.JobRequest.html#method.status
- Then update the job using update_with_id.

**Comment by JasterV** [comment]
Hello there :) As of 2026, the API of the library seems to have changed a lot and we can't find a way to kill/delete a scheduled job.

Can someone point us on the right direction? Thanks!

**Comment by geofmureithi** [comment]
What backend are yoi using?

**Reply by JasterV** [reply]
Postgres :)

**Comment by LeoBorai** [comment]
Hi @geofmureithi, I visited the link you shared: https://docs.rs/apalis/latest/apalis/prelude/struct.JobRequest.html#method.status
But seems like is no longer valid.

I think the way to go as of today is: https://docs.rs/apalis/latest/apalis/prelude/trait.Storage.html#tymethod.update?
